frontend/src/components/utils/Authentication.tsx

```diff
@@ -0,0 +1,42 @@
+import axios from 'axios';
+import { useEffect } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+
+const authenticatedRouts = ['/home'];
+const unauthenticatedRouts = ['/login', '/registration'];
+
+const Authentication = ({children}: any) => {
+    const history = useHistory();
+    const location = useLocation();
+
+    useEffect(() => {
+        try {
+            axios.get('http://localhost:3000/users').then(res => {
```

Comment:
You are calling the API on each page load, you should just check if token exists.